recipes/eleuther_eval.py

```diff
@@ -163,101 +440,84 @@ class EleutherEvalRecipe(EvalRecipeInterface):
-        self._limit = self._cfg.limit
-        self._tasks = list(self._cfg.tasks)
-        self._quantizer = config.instantiate(self._cfg.quantizer)
-        self._quantization_mode = training.get_quantizer_mode(self._quantizer)
-        self._enable_kv_cache = self._cfg.get("enable_kv_cache", True)
-
-        self._batch_size = self._cfg.batch_size
-        self._max_seq_length = self._cfg.get("max_seq_length", 4096)
-
-        training.set_seed(seed=self._cfg.seed)
-
-        checkpointer = config.instantiate(self._cfg.checkpointer)
-        if self._quantization_mode is None:
+        self.device = utils.get_device(device=cfg.device)
+        self.dtype = training.get_dtype(dtype=cfg.dtype, device=self.device)
+        self.logger = utils.get_logger(cfg.get("log_level", "info"))
+        training.set_seed(seed=cfg.seed)
+
+        # Eval specific variables
+        self.limit = cfg.limit
+        self.tasks = list(cfg.tasks)
+        self.max_seq_length = cfg.max_seq_length
+        self.batch_size = cfg.batch_size
+        self.enable_kv_cache = cfg.get("enable_kv_cache", True)
+        self.include_path = cfg.get("include_path", None)
+
+    def setup(self, cfg: DictConfig) -> None:
+        # Initialize quantizer and quantization mode
+        quantizer = config.instantiate(cfg.quantizer)
+        quantization_mode = training.get_quantizer_mode(quantizer)
+
+        # Load checkpoint
+        checkpointer = config.instantiate(cfg.checkpointer)
+        if quantization_mode is None:
             ckpt_dict = checkpointer.load_checkpoint()
         else:
             # weights_only needs to be False when loading a quantized model
             # currently loading a quantized model is only supported with the
             # FullModelTorchTuneCheckpointer
             ckpt_dict = checkpointer.load_checkpoint(weights_only=False)
 
-        self._model = self._setup_model(
-            model_cfg=self._cfg.model,
-            model_state_dict=ckpt_dict[training.MODEL_KEY],
-        )
-        self._tokenizer = config.instantiate(self._cfg.tokenizer)
-        logger.info("Tokenizer is initialized from file.")
+        # Initialize model
+        with training.set_default_dtype(self.dtype), self.device:
+            model = config.instantiate(cfg.model)
 
-    def _setup_model(
-        self,
-        model_cfg: DictConfig,
-        model_state_dict: Dict[str, Any],
-    ) -> nn.Module:
-        with training.set_default_dtype(self._dtype), self._device:
-            model = config.instantiate(model_cfg)
-
-        if self._quantization_mode is not None:
-            model = self._quantizer.quantize(model)
-            model = model.to(device=self._device, dtype=self._dtype)
-            for k, v in model_state_dict.items():
-                model_state_dict[k] = v.to(self._device)
-            model.load_state_dict(model_state_dict, assign=True)
-        else:
-            model.load_state_dict(model_state_dict)
```

Comment:
Was this changed deliberately? Based on #1403 I think we want to load state dict with assign=True when quantization is enabled

Reply:
Yeah.